jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/ctl/ActionSleep.java

```diff
@@ -50,45 +52,57 @@ public void validate(HttpAction action) {}
 
     @Override
     public void execute(HttpAction action) {
-        Runnable task = createRunnable(action);
+        SleepTask task = createRunnable(action);
         AsyncTask aTask = Async.execASyncTask(action, AsyncPool.get(), "sleep", task);
+        action.log.info(format("[%d] Sleep %d ms.", action.id, task.sleepMilli));
         JsonValue v = Async.asJson(aTask);
-        Async.setLocationHeader(action, aTask);
         ServletOps.sendJsonReponse(action, v);
     }
 
-    protected Runnable createRunnable(HttpAction action) {
+    protected SleepTask createRunnable(HttpAction action) {
         String interval = action.request.getParameter("interval");
         int sleepMilli = 5000;
-        if ( interval != null )
+        if ( interval != null ) {
             try {
                 sleepMilli = Integer.parseInt(interval);
             } catch (NumberFormatException ex) {
-                action.log.error(format("[%d] NumberFormatException: %s", action.id, interval));
+                ServletOps.errorBadRequest("Bad format for 'interval': integer required");
+                return null;
             }
-        action.log.info(format("[%d] Sleep %d ms", action.id, sleepMilli));
-        return new SleepTask(action, sleepMilli);
+        }
+        if ( sleepMilli < 0 ) {
+            ServletOps.errorBadRequest("Negative sleep interval");
+            return null;
+        }
+        if ( sleepMilli > MaxSleepMillis ) {
+            ServletOps.errorBadRequest("Sleep internal greater than maximum allowed");
+            return null;
+        }
+        return new SleepTask(action, sleepMilli, AsyncPool.get());
     }
 
     static class SleepTask implements Runnable {
         private final Logger log;
         private final long actionId;
-        private final int sleepMilli;
+        public  final int sleepMilli;
+        private final AsyncPool asyncPool;
 
-        public SleepTask(HttpAction action, int sleepMilli ) {
+        public SleepTask(HttpAction action, int sleepMilli, AsyncPool asyncPool ) {
             this.log = action.log;
             this.actionId = action.id;
             this.sleepMilli = sleepMilli;
+            this.asyncPool = asyncPool; 
         }
 
         @Override
         public void run() {
             try {
-                log.info(format("[%d] >> Sleep start", actionId));
-                Lib.sleep(sleepMilli);
-                log.info(format("[%d] << Sleep finish", actionId));
+                log.info(format("[Task %d] >> Sleep start", actionId));
+                for ( int i = 0 ; i < 10 ; i++ )
+                    Lib.sleep(sleepMilli/10);
```

Comment:
Out of curiosity, why `/ 10`? Is it to yield control so other things get a chance to run after every 1/10th of the sleep time?

Reply:
Yes - that's the reason. It is dev code put in to nudge the JVM along when running all in in JVM (client and Fuseki embedded server). I find that it sometimes shows up concurrency issues better and I was trying multiple "sleeps" running.

It isn't supposed to be there for the codebase and I'll remove it - good thing we have code review.